algorithms/integration/processor.py

```diff
@@ -499,7 +499,8 @@ def __call__(self):
 
             # I am 99% sure this is implied by all the code above
             assert (frame1 - frame0) <= len(imageset)
-            imageset = imageset[frame0:frame1]
+            if len(imageset) > 1:
+                imageset = imageset[frame0:frame1]
```

Comment:
It seems slightly concerning that this apparently results in an extra image read - is this the case for rotation data, or just still imagesets?

Reply:
I didn't profile `dials.integrate`, but I don't think it has the same problem.  I believe profile fitting is done from the strong shoeboxes, and data reading then only happens during integration itself.

In `dials.stills_process`, the same imageset is handed from program to program and so it cannot be reset.  In the `dials.integrate` it is invoked fresh and used once.  I believe :)

Reply:
Profile fitting is _not_ just done from the strong shoeboxes - for rotation data they are only used to determine the profile parameters - all reflections are then gathered to generate the reference profiles. 

We should probably check whether the rest is called for `dials.integrate` on rotation data as a separate exercise and raise an issue if so.